ci/do_ci.sh

```diff
@@ -85,21 +94,22 @@ if [[ "$1" == "bazel.release" ]]; then
   setup_clang_toolchain
   echo "bazel release build with tests..."
   bazel_binary_build release
+  bazel_release_options="test ${BAZEL_TEST_OPTIONS} -c opt ${MEMORY_TEST_EXACT_ARGS}"
 
   if [[ $# -gt 1 ]]; then
     shift
     echo "Testing $* ..."
     # Run only specified tests. Argument can be a single test
     # (e.g. '//test/common/common:assert_test') or a test group (e.g. '//test/common/...')
-    bazel_with_collection test ${BAZEL_TEST_OPTIONS} -c opt $*
+    bazel_with_collection $bazel_release_options $*
```

Comment:
@htuch @lizan I could use your help here. The script I have here doesn't work currently on my system due to docker issues. I injected a bug in the latest commit and expected 'release' to fail with a wrong-byte-count expect-failure, but it passed.

Any specific advice on how to repro CI locally or debug CI would be helpful.

Reply:
AFAIK `release` doesn't fail because `hasDeterministicMallocStats` return false for optimized build, so that's what you should fix.

Reply:
Great catch; I knew I was missing something. It must be that the optimizer is removing my call to new an array into a unique_ptr that is never again referenced.  Thanks Lizan.

Reply:
Yeah I guess that's the case. In last few PRs that adjusts the exact match, only coverage and ipv6_tests were failing.